src/odbc/src/app/application_data_buffer.cpp

```diff
@@ -319,7 +319,7 @@ ConversionResult::Type ApplicationDataBuffer::PutStrToStrBuffer(
   // resLenPtr will receive the remaining required data, gradually decreasing in length as
   // more calls with the same column are made.
   SqlLen remainingBytesRequired = bytesRequired - totalBytesWritten > 0 ?
-    static_cast<SqlLen>(bytesRequired - bytesWritten) : bytesRequired;
+    static_cast<SqlLen>(bytesRequired - totalBytesWritten) : bytesRequired;
```

Comment:
`bytesWritten` is the number of bytes written for the current call. Obviously, using this instead of `totalBytesWritten` would not decrease `remainingBytesRequired` properly.